test/testenv/appframework_utils.go

```diff
@@ -60,7 +63,13 @@ func GetPodAppStatus(deployment *Deployment, podName string, ns string, appname
 func GetPodInstalledAppVersion(deployment *Deployment, podName string, ns string, appname string) (string, error) {
 	path := "etc/apps"
 	if strings.Contains(podName, "-indexer-") {
-		path = "etc/slave-apps/"
+		path = "etc/slave-apps"
+	}
+	if strings.Contains(podName, "cluster-master") {
+		path = "etc/master-apps"
+	}
+	if strings.Contains(podName, "-deployer-") {
+		path = "etc/shcluster/apps"
```

Comment:
I would rather use a `switch`or corresponding `else`s here instead of just `if` blocks since you only need one condition to be true. Right now it will check all the `if`s which is unnecessary.